src/PhpParser/Printer/BetterStandardPrinter.php

```diff
@@ -102,6 +114,12 @@ public function printFormatPreserving(array $stmts, array $origStmts, array $ori
 
         $content = parent::printFormatPreserving($newStmts, $origStmts, $origTokens);
 
+        // strip empty starting/ending php tags
+        if (array_key_exists(0, $stmts) && $stmts[0] instanceof FileWithoutNamespace) {
+            $content = Strings::replace($content, self::EMPTY_STARTING_TAG_REGEX, '');
+            $content = Strings::replace(\rtrim($content), self::EMPTY_ENDING_TAG_REGEX, '', ) ."\n";
```

Comment:
both match should be checked first before replace:

```php
use Rector\Core\Util\StringUtils;

if (StringUtils::isMatch($content, self::EMPTY_STARTING_TAG_REGEX) && StringUtils::isMatch($content, self::EMPTY_ENDING_TAG_REGEX)) {
            $content = Strings::replace($content, self::EMPTY_STARTING_TAG_REGEX, '');
            $content = Strings::replace(\rtrim($content), self::EMPTY_ENDING_TAG_REGEX, '', ) ."\n";
}
```

to ensure both regex checked, not just one.

Reply:
hmm do you have a concrete case in mind in which this matters.

IMO a file could also just contain a starting or ending start-tag which should be stripped independently

Reply:
It ensure there is no empy `<?php` just replaced with empty string, see https://regex101.com/r/CBldvN/1 that empty `<?php` getted with the your `self::EMPTY_ENDING_TAG_REGEX` regex.

Reply:
I think we would need to come up with a test-case for the mentioned situation.
this should be put in a separate file within plain-views/src

I am not yet sure how a file need to look like to be affected

Reply:
You can add 1 file like empty open php file is needed: 

``` // empty-open-php-file.php
<?php
```

ensure it not affected.

Reply:
another solution is update the regex value with `ms`

```php
private const EMPTY_ENDING_TAG_REGEX = '#.{1,}(<\?php)$#ms';
```

see https://regex101.com/r/bO7h2V/1

Reply:
added a empty file like you suggested. the fix, without the suggested change is still green